spec/use_packs_spec.rb

```diff
@@ -24,14 +24,45 @@ def expect_files_to_not_exist(files)
   end
 
   describe '.create_pack!' do
-    # Right now, `UsePacks` only supports `packs`, `gems`, or `components` as the home for packwerk packages
-    context 'pack name does not include `packs` prefix' do
-      let(:pack_name) { 'my_pack' }
+    context 'when overriding the allow packs paths' do
+      before do
+        stub_const("UsePacks::PERMITTED_PACK_LOCATIONS", %w[foo/*])
```

Comment:
Can you try again with `write_file` and let me know what errors you run into? We're trying to have our tests only do setup and assertions using public API as much as possible. In this case, this constant is not part of public API and in fact I think we can probably remove it entirely since I'm not sure it's needed anymore.

If you're running into issues writing `packs.yml` because of load order or anything let me know and I can help debug, as I want to make it as straightforward as possible to write tests 🙏 

Note that `use_pack` has this line in it's `spec_helper.rb`:
```
require 'packs/rspec/support'
```

That file in `packs` has a `Packs.bust_cache!` before every test, which should allow us to rewrite `packs.yml` without issue 🤞